Assets/SteamVR_Unity_Toolkit/Scripts/Controls/2D/RadialMenu/RadialMenu.cs

```diff
@@ -0,0 +1,300 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.Events;
+using System.Collections.Generic;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+
+[ExecuteInEditMode] //Lets us set up buttons from inspector option
+public class RadialMenu : MonoBehaviour {
+
+    #region Variables
+    public List<RadialMenuButton> Buttons;
+	public GameObject buttonPrefab;
+	[Range(0f,1f)]
+	public float buttonThickness = 0.5f;
+	public Color buttonColor = Color.white;
+	public float offsetDistance = 1;
+	[Range(0,359)]
+	public float offsetRotation;
+
+	public bool rotateIcons;
+	public float iconMargin;
+    public bool isShown;
+    public bool HideOnRelease;
+
+    //Has to be public to keep state from editor -> play mode?
+	public List<GameObject> menuButtons;
+
+    float currentTouchAngle;
+    int currentHover = -1;
+    int currentPress = -1;
+    #endregion
+
+    #region Unity Methods
+
+    void Start()
+    {
+        if (Application.isPlaying)
+        {
+            if(!isShown)
+                transform.localScale = Vector3.zero;
+            RegenerateButtons();
+        }
+            
```

Comment:
nitpicking the redundant extra line :) may as well remove it if you're in here doing the auto formatting stuff